src/scheduler/pbs_sched.cpp

```diff
@@ -1199,53 +1084,45 @@ main(int argc, char *argv[])
 
 	pthread_mutex_init(&cleanup_lock, &attr);
 
-	connect_servers();
+	sched_svr_init();
+	connect_svrpool();
 
 	for (go=1; go;) {
+		int i;
 
 		wait_for_cmds();
-		while (go && !ds_queue_is_empty(sched_cmds)) {
-			sched_cmd *cmd = static_cast<sched_cmd *>(ds_dequeue(sched_cmds));
-			sched_svrconn *sconn = NULL;
 
-			if (!cmd)
-				continue;
-
-			sconn = find_server(cmd->from_sock);
-			if (!sconn) {
-				/* shouldn't happen, but if it does just continue */
-				free_sched_cmd(cmd);
-				continue;
+		/* First walk through the qrun_list as this has high priority followed by the other commands */
+		for (i = 0; i < qrun_list_size; i++) { 
+			if (schedule_wrapper(&qrun_list[i], opt_no_restart) == 1) {
+				go = 0;
+				break;
 			}
+			free(qrun_list[i].jid);
+			qrun_list[i].jid = NULL;
+		}
 
-#ifdef PBS_UNDOLR_ENABLED
-			if (sigusr1_flag)
-				undolr();
-#endif
-			if (sigprocmask(SIG_BLOCK, &allsigs, &oldsigs) == -1)
-				log_err(errno, __func__, "sigprocmask(SIG_BLOCK)");
+		for (i = 0; go && (i < SCH_CMD_HIGH); i++) {
+			sched_cmd cmd;
 
-			/* Keep track of time to use in SIGSEGV handler */
-			now = time(NULL);
-			if (!opt_no_restart)
-				segv_last_time = now;
-#ifdef DEBUG
-			{
-				strftime(log_buffer, sizeof(log_buffer), "%Y-%m-%d %H:%M:%S", localtime(&now));
-				DBPRT(("%s Scheduler received command %d\n", log_buffer, sconn->cmd));
-			}
-#endif
+			if (sched_cmds[i] == 0)
```

Comment:
Since '0' is a valid command, I'd set cmd to -1 when it is processed and check for it to be not -1 before calling the wrapper.

Reply:
Same as my earlier reply.